### DISCUSSION TITLE
How can I access the history of changes in the fluid framework?

### DISCUSSION THREAD

**Discussion by Sassyprogrammer1** [microsoft/FluidFramework#8616] [General]
I see the FAQ section of the fluid framework says that, "Yes. Fluid inherently keeps all changes and these are accessible through the framework". So can someone shed a little bit of light on this like how can I actually access the history of changes?

**Comment by tylerbutler** [comment]
The docs are a little outdated here. The op history and summaries are accessible through Fluid, but there's no public API to work with them. This could change, so I'd love to know what you are hoping to do with access to the raw op history or summaries.

Also, the next lines of the FAQ read, "The only caveat is that for performance and storage efficiency, operations need to be summarized from time to time. This may cause a loss of granularity." This is referring to the fact that a Fluid service is not _required_ to retain all ops forever -- rather, they have to keep enough summaries and enough ops that clients can rebuild their local states if needed. So in practice, the entire op history is not available other than in summarized form.
